onadata/apps/api/viewsets/data_viewset.py

```diff
@@ -364,8 +370,23 @@ def destroy(self, request, *args, **kwargs):
                     deleted_at__isnull=True,
                 )
 
+            error_msg = None
             for instance in queryset.iterator():
-                delete_instance(instance, request.user)
+                if permanent_delete:
+                    if enable_submission_permanent_delete:
+                        instance.delete()
+                    else:
+                        error_msg = {
```

Comment:
I think we should move this out of the loop else `error_msg` will be set for each instance